### PR TITLE
Fix a bug in data_size when only one line in block

### PR DESCRIPTION
**Pull request by uzadude**
## Summary

We had a "special" treatment for the case when Avro block contains only one line.
seems it has a bug when it happens on the first line.

resolves #51

## Detailed Description
removing this special case seems to resolve the issue.

## How was it tested?
added a unit test

### CODE REVIEW

File: dione-hadoop/src/main/scala/com/paypal/dione/hdfs/index/avro/AvroIndexer.scala [hunk 1]
```diff
@@ -63,21 +63,15 @@ case class AvroIndexer(file: Path, start: Long, end: Long, conf: Configuration)
...
     lastPosition = fileReader.previousSync()
     HdfsIndexerMetadata(file.toString, prevSync, numInBlock, size)
   }
 
   def getSchema(): Schema = fileReader.getSchema
 
-  private def tryInferSize(reader: DataFileReader[GenericRecord], lastPosition: Long, totalInBlock: Int): Int = {
-    val pos = reader.previousSync()
-    val longSize =
-      if (totalInBlock == 1) // exactly 1 row in block
```
Comment (uzadude) [inline dione-hadoop/src/main/scala/com/paypal/dione/hdfs/index/avro/AvroIndexer.scala:76]: @shay1bz - do you happen to remember why we had this in the first place?

Comment (shay1bz) [reply dione-hadoop/src/main/scala/com/paypal/dione/hdfs/index/avro/AvroIndexer.scala:76]: I think I remember that dividing block size with block count (as in PR) failed on some edge case - but I really don't remember which =\ maybe first/last block or something like that

File: dione-spark/src/test/scala/com/paypal/dione/spark/index/avro/TestOneLineInBlock.scala [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+
+  override val baseTestPath: String = "TestData/TestOneLineInBlock/"
+  override val dbName: String = "TestOneLineInBlock"
+
+  @BeforeAll
+  def initData(): Unit = {
+    import spark.implicits._
+
+    //    val sc = spark.sparkContext
+    //    sc.hadoopConfiguration.setInt("avro.mapred.sync.interval", 100)
```
Comment (shay1bz) [inline dione-spark/src/test/scala/com/paypal/dione/spark/index/avro/TestOneLineInBlock.scala:24]: why is that commented out? the test should have interval=1, no?

Comment (uzadude) [reply dione-spark/src/test/scala/com/paypal/dione/spark/index/avro/TestOneLineInBlock.scala:24]: unfortunately, it doesn't work so instead of decreasing the sync.interval i increased the data size to be more than 64k (default value)

**Review by shay1bz — APPROVED** [review]
